src/humanloop/evals/run.py

```diff
@@ -493,69 +631,66 @@ def _file_or_file_inside_hl_utility(file: File) -> File:
+def _check_file_type(file_config: FileEvalConfig) -> FileEvalConfig:
+    """Check that the file type is provided, or set it to `flow` if not provided."""
     try:
-        type_ = typing.cast(FileType, file.pop("type"))  # type: ignore [arg-type, misc]
-        sys.stdout.write(
-            f"{CYAN}Evaluating your {type_} function corresponding to `{file.get('path') or file.get('id')}` on Humanloop{RESET}\n\n"
+        type_ = typing.cast(FileType, file_config.pop("type"))  # type: ignore [arg-type, misc]
+        print_info(
+            f"Evaluating your {type_} function corresponding to `{file_config.get('path') or file_config.get('id')}` on Humanloop\n\n"
         )
-        return type_ or "flow"
+        file_config["type"] = type_ or "flow"
     except KeyError as _:
         type_ = "flow"
-        sys.stdout.write(f"{YELLOW}No `file` type specified, defaulting to flow.{RESET}\n")
-        return type_
+        print_warning("No `file` type specified, defaulting to flow.")
+        file_config["type"] = type_
+    return file_config
 
 
-def _get_file_callable(file: File, type_: FileType) -> Optional[Callable]:
-    # Get the `callable` from the `file` to Evaluate
-    function_ = typing.cast(Optional[Callable], file.pop("callable", None))
+def _get_file_callable(file_config: FileEvalConfig) -> Optional[Callable]:
+    """Get the callable of the File to be evaluated, or throw if None was provided for Flows."""
+    type_ = file_config.get("type")
+    function_ = typing.cast(Optional[Callable], file_config.pop("callable", None))
     if function_ is None:
         if type_ == "flow":
-            raise ValueError("You must provide a `callable` for your Flow `file` to run a local eval.")
+            raise HumanloopRuntimeError("You must provide a `callable` for your Flow `file` to run a local eval.")
         else:
-            sys.stdout.write(
-                f"{CYAN}No `callable` provided for your {type_} file - will attempt to generate logs on Humanloop.\n\n{RESET}"
+            print_info(
+                f"No `callable` provided for your {type_} file - will attempt to generate logs on Humanloop.\n\n"
             )
+    elif type_ == "agent":
+        raise ValueError("Agent evaluation is only possible on the Humanloop runtime, do not provide a `callable`.")
     return function_
 
 
-def _upsert_file(
-    file: File, type: FileType, client: "BaseHumanloop"
-) -> Union[PromptResponse, FlowResponse, ToolResponse, EvaluatorResponse]:
+def _upsert_file(client: "BaseHumanloop", file_config: FileEvalConfig) -> EvaluatedFile:
     # Get or create the file on Humanloop
-    version = file.pop("version", {})
-    file_dict = {**file, **version}
+    version = file_config.pop("version", {})
+    file_dict = {**file_config, **version}
+    del file_dict["type"]
+    type_ = file_config.get("type")
+    subclient = _get_subclient(client=client, file_config=file_config)
 
-    if type == "flow":
+    if type_ == "flow":
```

Comment:
no agent here? Why can't you upsert? From what I can tell this upserts based on the version param in eval.run?